classes/helper/HelperOptions.php

```diff
@@ -102,6 +102,13 @@ public function generateOptions($option_list)
 
                 $field['required'] = isset($field['required']) ? $field['required'] : $this->required;
 
+                if ($field['type'] === 'checkbox' && isset($field['value_multiple'])) {
```

Comment:
I'd add a little comment here to explain why this new condition is added:

```suggestion
                // this is for a field made of multiple checkboxes
                if ($field['type'] === 'checkbox' && isset($field['value_multiple'])) {
```

Reply:
I understand introducing a multiple-checkbox field, although it wasn't necessary, and it, IMHO, expanded the PR scope. Right now, we need to think it's a good implementation providing good DX. You could:

-  render the field as wished with the override coming from the controller, you could have `image_format_selector` field
- process the options as you wish by overriding the implementation of `processOptions`

Reply:
I'll have a look into this :+1:  we struggled with @nicosomb when playing with `processOptions` but we might have missed something

Reply:
About the multiple checkbox, it was what was requested by product team

Reply:
@matthieu-rolland, just to be clear, I understand the need to have them but I don't like that because of this, there are changes introduced in the core that basically introduce new feature for developers which maybe isn't implemented optimally

Reply:
I've had a look at checkboxes in legacy BO forms

What I've found is that multiple checkboxes are already supported even if we didn't use it so far:

- checkbox types can have a `choices` attribute, which takes a list of key-value (name => label) checkboxes

- in the template, there's a foreach on `choices`, to display the multiple checkboxes

The only thing that was added here is `value_multiple`, which is probably badly named, we could call it `initial_values`, I think it is necessary for everyone wanting to use multiple checkboxes.

I'm not sure I understand what you wanted to do instead ?

Reply:
I guess this is  a requirement if we want to use a multiple checkbox input in this form I'm actually surprized the problem was never raised as a bug because it seems like the HelperForm never allowed multiple checkboxes, even with it uses checkbox as the input for appears to be a single choice input so it should rely on a radio input The UX here really doesn't seem good

Although changing the initial single input to radio may be too important of a change for a minor version. However adding this new option doesn't have that much impact since the default value is false so it won't change the existing forms it's fine by me

Reply:
Although I would have used a different naming like `multiple` or `multiple_choices` but not blocking I guess